### ISSUE TITLE
Cannot get typing's to Lint at all. 

### ISSUE THREAD

**Issue by jtwigg** [microsoft/pyright#1442]
**Describe the bug**
I'm having trouble getting even basic type checking to lint.
I've got "Python" VS Code plugin installed and "PyLance" VS Code plugin installed.
`"python.languageServer": "Pylance"`
is enabled too.

PyLance does report some linting errors, but non that the generic plugin wasn't already doing.

**To Reproduce**
```python
def test(x: int) -> int:
    x = x + 1
    return "should fail"

y: str = test(x="should also fail")

123: # Does report Pylance error so it's enabled at least.
```

**Expected behavior**
I would expect both the return type of `test` function to fail as it should be returning an `int`
I would expect the call into `test` to fail as it should be passing in an `int` and not an `str`
I would expect the return type assignment to `y:str` to fail because it would be receiving an `int` from the function.

**VS Code extension or command-line**
PLance `v2021.1.3` VS Code

**Comment by erictraut** [comment]
By default, type checking errors are not enabled in Pylance because most users are not interested in such errors. You can enable basic type checking by setting "python.analysis.typeCheckingMode" to "basic".

**Comment by jtwigg** [comment]
@erictraut Amazing! Any other things you suggest I should turn on?

**Comment by erictraut** [comment]
Pylance is built on the open-source [pyright](https://github.com/microsoft/pyright) type checker, which offers many advanced features that are disabled by default. For a full list, see [this documentation](https://github.com/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md). If you want to enable strict rules, you can set the type checking mode to "strict". [This table](https://github.com/microsoft/pyright/blob/master/docs/configuration.md#diagnostic-rule-defaults) shows which diagnostic rules are enabled by default in each of the three type checking modes (off, basic and strict).
